src/network/networkd-link.c

```diff
@@ -1972,8 +1982,14 @@ static int link_joined(Link *link) {
                 r = link_set_bridge(link);
                 if (r < 0)
                         log_link_error_errno(link, r, "Could not set bridge message: %m");
+
         }
 
+        /* TODO: check for bridge interface? */
```

Comment:
I think we can track it via via netlink messages what kernel feeds us. Parse it and look for IFLA_MASTER. Again it could be a bond so we need to look for bridge properties are set .